packager/launchEditor.js

```diff
@@ -30,19 +50,22 @@ function launchEditor(fileName, lineNumber) {
     return;
   }
 
-  var argument = fileName;
+  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
+  var args = [fileName];
   if (lineNumber) {
-    argument += ':' + lineNumber;
+    args = getArgumentsForLineNumber(editor, fileName, lineNumber);
   }
 
-  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
   if (editor) {
     console.log('Opening ' + chalk.underline(fileName) + ' with ' + chalk.bold(editor));
-    exec(editor + ' ' + argument, function(error) {
-      if (error) {
-        console.log(chalk.red(error.message));
-        printInstructions('How to fix');
+
+    var child = spawn(editor, args, {stdio: 'inherit'});
```

Comment:
This is for terminal editors? They might cause a mess since the packager could keep writing to stdout while the editor is trying to write to stdout too, although maybe it's fine in practice.